### DISCUSSION TITLE
Fail wdpost when using ZFS

### DISCUSSION THREAD

**Discussion by de-authority** [filecoin-project/lotus#7628] [Storage Provider]
We have a [bounty](https://github.com/filecoin-project/community/issues/318) about the best configuration of ZFS. It is in progress. there will be good documents for reference soon. Before that, some community members will encounter the failure of  wdpost when taking zfs as the background storage, which usually occurs on the 2nd Sunday of the month.

The following methods are worth trying:
~~Edit `/etc/cron.d/zfsutils-linux` and comment this line out~~

According to @stuberman's experience, you can try the following

1: To comment this line in `/etc/cron.d/zfsutils-linux` directly, you need to manually scrub

```shell
24 0 8-14 * * root if [ $(date +\%w) -eq 0 ] && [ -x /usr/lib/zfs-linux/scrub ]; then /usr/lib/zfs-linux/scrub; fi
```
2: Modify the cycle and time of scheduled tasks in  `/etc/cron.d/zfsutils-linux` to reduce the impact on wdpost

3: try this solution: https://github.com/filecoin-project/lotus/discussions/7522

**Comment by stuberman** [comment]
Storage providers using ZFS will have corruption problems if they do not scrub on a regular basis. I like to scrub every 7 days with SATA drives, SAS drives can tolerate monthly scrubs. You can schedule a scrub or manually scrub. To protect the wdpost you can change the priority, see this [thread](https://github.com/filecoin-project/lotus/discussions/7522).

**Reply by de-authority** [reply]
Thanks a lot. Update the post .how about now?
